upload_artifacts.sh

```diff
@@ -1,21 +1,33 @@
 #!/bin/bash
 
-# Used in .travis.yml
+# This script is used in .travis.yml
+
+set -e
 
 BUCKET=piksi-buildroot-images
 PIKSI_VERSION=v3
+BUILD_VERSION=$(git rev-parse --short HEAD)
 
-set -e
+if [ "$TRAVIS_PULL_REQUEST" == "true" ]; then
+  FOLDER=pull_requests
+else
+  if [ "$TRAVIS_BRANCH" == "master" ]; then
+    FOLDER=master
+  else
+    FOLDER=misc
+  fi
+fi
 
-BUILD_VERSION=$(git rev-parse --short HEAD)
-BUILD_DIR=commit_$BUILD_VERSION
-# "folders" on S3 are prefix keys
-# http://stackoverflow.com/questions/33113354/how-to-create-a-folder-like-i-e-prefix-object-on-s3-using-the-aws-cli
-aws s3api put-object --bucket $BUCKET --key $PIKSI_VERSION/$BUILD_DIR/
-UPLOAD_DIR="s3://$BUCKET/$PIKSI_VERSION/$BUILD_DIR"
+BUILD_DIR="UTC-$(date -u +%T)_TRAVIS-$(echo $TRAVIS_BUILD_NUMBER)_COMMIT-$(echo $BUILD_VERSION)"
+
+mkdir -p uploads/$BUILD_DIR
+
+UPLOAD_DIR="s3://$BUCKET/$PIKSI_VERSION/$FOLDER/"
+
+cp "./buildroot/output/images/boot.bin" "./uploads/$BUILD_DIR"
```

Comment:
Ahhhh you're right, I don't need that. For some reason I thought that uploading like `aws s3 cp --recursive yourfile.txt s3://bucket/prefixthatdoesntexistyet/` then that wouldn't work but it actually does 😑